$OpenBSD: patch-src_plugins_qmakeprojectmanager_qmakebuildconfiguration_cpp,v 1.6 2018/12/08 15:08:58 rsadowski Exp $

When calling gmake, override the CXX and LINK variables in Makefiles generated
by qmake by setting gmake's command line arguments. Makes it possible to
compile C++11 code.

Index: src/plugins/qmakeprojectmanager/qmakebuildconfiguration.cpp
--- src/plugins/qmakeprojectmanager/qmakebuildconfiguration.cpp.orig
+++ src/plugins/qmakeprojectmanager/qmakebuildconfiguration.cpp
@@ -129,6 +129,7 @@ void QmakeBuildConfiguration::initialize(const BuildIn
 
     BuildStepList *buildSteps = stepList(ProjectExplorer::Constants::BUILDSTEPS_BUILD);
     auto qmakeStep = new QMakeStep(buildSteps);
+    qmakeStep->setUserArguments(QLatin1String("CXX=eg++ LINK=eg++"));
     buildSteps->appendStep(qmakeStep);
     buildSteps->appendStep(new QmakeMakeStep(buildSteps));
 
